### PR TITLE
Capture JFR recording for compiler daemons

### PR DESCRIPTION
**Pull request by wolfs**
To see what is going on in the compiler daemons of `largeMonolithicJavaProject`.

### CODE REVIEW

**Comment by wolfs** [comment]
Here is an example build capturing the compiler daemon JFR recordings: https://builds.gradle.org/viewLog.html?buildId=21788242&

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JfrProfiler.groovy [hunk 1]
```diff
@@ -59,11 +59,15 @@ class JfrProfiler extends Profiler implements Stoppable {
...
     List<String> getAdditionalJvmOpts(BuildExperimentSpec spec) {
-        String flightRecordOptions = "stackdepth=1024"
         def jfrFile = getJfrFile(spec)
         jfrFile.parentFile.mkdirs()
-        if (!useDaemon(spec)) {
-            flightRecordOptions += ",defaultrecording=true,dumponexit=true,dumponexitpath=${jfrFile},settings=$config"
+        getJvmOpts(!useDaemon(spec), jfrFile)
+    }
+
+    private List<String> getJvmOpts(boolean startRecordingImmediately, File jfrOutputLocation) {
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JfrProfiler.groovy]: Why this is named `startRecordingImmediately`? `startRecordingImmediately` seems to imply "there would be recording later", but looking at the code, it's controlling whether to enable recording or not. WDYT?

Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JfrProfiler.groovy]: If you don't start the recording immediately, then you need to use the `start`/`stop` method to start the recording. The JVM parameter controls exactly if the recording starts when the JVM starts or not.

Comment (blindpirate) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JfrProfiler.groovy]: OK, these are JFR params, not `-XX:+UnlockCommercialFeatures -XX:+FlightRecorder` which needs to be set up at JVM startup. I mixed them up. thanks for explaining.

**Pull request merged**